### DISCUSSION TITLE
Facing some issues with the WebView control in the pre-release version(0.25.0.dev3717) on Web

### DISCUSSION THREAD

**Discussion by Benitmulindwa** [flet-dev/flet#4356] [General]
@ndonkoHenri Here we are!

I can not update the value of the *url* property, EXPLANATIONS: let's say i want to update the WebView's *url* property when the button is cliked, am getting an error(`This method is supported on Androatforms only.`), but I can change its value when am not using an function/eventhandler. ```wv.url="https://flet.dev" if a==1 else "https://example.com"```

Example:
```python


 wv = WebView(
         url="https://flet.dev",
         expand=True,
         enable_javascript=True,
       
    )
  def update_url():
     wv.url="https://example.com"
     wv.update()
  
   page.add(wv)
   page.add(
        ElevatedButton(
            "Update Url",
            on_click=lambda e: update_url()),
        )
    )

```

**Comment by ndonkoHenri** [comment]
On which platform are you running the app? Can you share runnable code repro for both situations (working and failing)?

**Reply by Benitmulindwa** [reply]
am running it on a windows pc but as a web app

```python
from flet import *


def main(page: Page):
    page.padding = 0
    
    def update_url():
         wv.url="https://flet.dev/docs/controls/webview/"
         wv.update()

    wv = WebView(
        url="https://flet.dev",
        expand=True,
        enable_javascript=True,
        
    )
   

    page.add(wv)
    page.add(
        ElevatedButton(
            "Update Rotation",
            on_click=lambda e: update_url(),
        )
    )


app(main)

```
Output:
`This method is supported on Android, iOS and macOS platforms only.`

**Reply by Benitmulindwa** [reply]
am not able to update the almost all the `WebView`'s properties, and if doable, plz make the `run_javascript()` method available on web as well and the events(`on_url_change`,  `on_console_message`, etc)
